packages/desktop-client/src/components/reports/spreadsheets/filterHiddenItems.ts

```diff
@@ -6,19 +6,46 @@ import {
 export function filterHiddenItems(
   item: UncategorizedEntity,
   data: QueryDataEntity[],
+  showOffBudget?: boolean,
+  showHiddenCategories?: boolean,
+  showUncategorized?: boolean,
 ) {
-  return data.filter(asset => {
+  const showHide = data
+    .filter(e =>
+      !showHiddenCategories
+        ? e.categoryHidden === false && e.categoryGroupHidden === false
+        : true,
+    )
+    .filter(f =>
```

Comment:
Because uncategorized transactions and offbudget and transfers share so many similarities I needed a way to figure out how/when to display each item. This filter check the toggle status of each filter and displays data according to those states.